object_detection/download_dataset.sh

```diff
@@ -1,20 +1,33 @@
 #!/bin/bash
 
 # Get COCO 2017 data sets
-mkdir -p pytorch/datasets/coco
-pushd pytorch/datasets/coco
+DATA_ROOT_DIR="${DATA_ROOT_DIR:-./pytorch/datasets/coco}"
+echo "Downloaading to folder: $DATA_ROOT_DIR"
+mkdir -p $DATA_ROOT_DIR
+pushd $DATA_ROOT_DIR
 
+echo "Downloading coco_annotations_minival.tgz:"
 curl -O https://dl.fbaipublicfiles.com/detectron/coco/coco_annotations_minival.tgz
-tar xzf coco_annotations_minival.tgz
+echo "Extracting coco_annotations_minival.tgz:"
+tar -xzf coco_annotations_minival.tgz &>/dev/null
 
-curl -O http://images.cocodataset.org/zips/train2017.zip
-unzip train2017.zip
+echo "Downloading annotations_trainval2017.zip:"
+curl -O http://images.cocodataset.org/annotations/annotations_trainval2017.zip
+echo "Extracting annotations_trainval2017.zip:"
+n_files=`unzip -l  annotations_trainval2017.zip| grep .json | wc -l`
+unzip annotations_trainval2017.zip  | { I=-1; while read; do printf "Progress: $((++I*100/$n_files))%%\r"; done; echo ""; }
 
+echo "Downloading val2017.zip:"
 curl -O http://images.cocodataset.org/zips/val2017.zip
-unzip val2017.zip
+echo "Extracting val2017.zip:"
+n_files=`unzip -l  val2017.zip| grep .jpg | wc -l`
+unzip val2017.zip  | { I=-1; while read; do printf "Progress: $((++I*100/$n_files))%%\r"; done; echo ""; }
 
-curl -O http://images.cocodataset.org/annotations/annotations_trainval2017.zip
-unzip annotations_trainval2017.zip
+echo "Downloading train2017.zip:"
+curl -O http://images.cocodataset.org/zips/train2017.zip
+echo "Extracting train2017.zip:"
+n_files=`unzip -l  train2017.zip| grep .jpg | wc -l`
+unzip train2017.zip  | { I=-1; while read; do printf "Progress: $((++I*100/$n_files))%%\r"; done; echo ""; }
 
 # TBD: MD5 verification
 # $md5sum *.zip *.tgz
```

Comment:
Can you please add a checksum verification step to make sure changes do not affect the dataset?

Reply:
Done, added the validation inside the download_dataset.sh file.